Marlin/src/module/probe.cpp

```diff
@@ -348,6 +348,34 @@ bool Probe::set_deployed(const bool deploy) {
     constexpr bool deploy_stow_condition = true;
   #endif
 
+  #if BOTH(NOZZLE_AS_PROBE, PROBE_REQUIRES_MINTEMP)
+    bool setting_hotend, setting_bed = false;
```

Comment:
setting_hotend isn't initialized, and could call into the thermalManager incorrectly.

You can also move each of those into their respective `#if` statements.

Reply:
Moving them inside compiler errored on the second set of checks with the wait statement.